enhancements/operator-conditions-lib.md

```diff
@@ -49,40 +49,99 @@ Operators need to be able to read the Status CR created by OLM, and also have th
 
 #### Story 1
 
-As an operator author, I want my operator to be able to add/modify conditions in the `Status` CR.
-
-#### Story 2
-
-As an operator author, I would like to know the `NamespacedName` of the `Status` CR, which can be used to obtain the resource from cluster.
+As an operator author, I want my operator to be able to get/update conditions in the `OperatorCondition` CR.
 
 ## Design Details
 
-When an operator is installed in the cluster, OLM will create a `Status` Custom Resource for the operator. The operator will be able to perform the following actions using the APIs provided in the library:
+When an operator is installed in the cluster, OLM will create an `OperatorCondition` Custom Resource for the operator. The user will be expected to provide a [client][cr_client] using which the library can access the operator condition resource present in the cluster, owned by the operator. The library will provide an interface, using which users can get the required condition from the CRD, or can set the status of the specific condition.
+
+There would be a generic operatorCondition manager interface, which would in turn have the methods to `get` and `set` a [conditionType][condition_type_api]. For example:
 
-1. Get the `NamespacedName` of the CRD:
+```go
+type ConditionManager interface {
+  // GetCondition gets the condition on the operator's 
+	// OperatorCondition. It returns an error if there are problems getting
+	// the OperatorCondition object or if the specific condition type does not
+	// exist.
+  GetCondition(context.Context, metav1.ConditionType) (*metav1.Condition, error)
+
+  // SetConditionStatus sets the specific condition on the operator's 
+	// OperatorCondition to the provided status. It returns an error if there
+	// are problems getting or updating the OperatorCondition object.
+  SetConditionStatus(context.Context, metav1.ConditionType, metav1.ConditionStatus, ...Option) error
+
+  // GetNamespacedName returns the NamespacedName of the CR. It returns an error
+  // when the name of the CR cannot be found from the environment variable set by
+  // OLM, or when the namespace cannot be found from the associated service account
+  // secret.
+  GetNamespacedName(context.Context) (types.NamespacedName, error)
```

Comment:
Looking into this, I realized that with this library we cannot provide users the ability to add new conditions. That was one of the deliverable in the previous proposal. Even if we direct users to utilize upstream conditions, they would need a way to know the `NamespacedName`. There are two options which we can do:

1. Expose the NamespacedName. 
If we do this, as an operator writer, I would have to first use this library to get the `NamespacedName`. Then use upstream helpers to add the condition. And then use controller-runtime's client.Update. The second option is:

2. In `SetConditionStatus` if the conditionType is not present, we create a new condition:
If we do this, we may have to know the entire condition object which the user wants to add, instead of knowing just the conditionType. I suppose, this may not be the right thing to do in `SetConditionStatus` at least.

If option 1 is what we want to follow, it sounds a bit convoluted instead of being user friendly. Is that fine, @joelanford and @estroz ?